packages/@lwc/ssr-compiler/src/transmogrify.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { traverse, builders as b, type NodePath } from 'estree-toolkit';
+import { produce } from 'immer';
+import type { Node } from 'estree';
+import type { Program as EsProgram } from 'estree';
+
+export type TransmogrificationMode = 'sync' | 'async';
+
+interface TransmogrificationState {
+    mode: TransmogrificationMode;
+}
+
+export type Visitors = Parameters<typeof traverse<Node, TransmogrificationState>>[1];
+
+const EMIT_IDENT = b.identifier('$$emit');
+const TMPL_FN_PATTERN = /tmpl($\d+)?/;
+const GEN_MARKUP_PATTERN = /generateMarkup.*/;
+
+const isWithinFn = (pattern: RegExp, nodePath: NodePath): boolean => {
+    const { node } = nodePath;
+    if (!node) {
+        return false;
+    }
+    if (node.type === 'FunctionDeclaration' && pattern.test(node.id.name)) {
+        return true;
+    }
+    if (nodePath.parentPath) {
+        return isWithinFn(pattern, nodePath.parentPath);
+    }
+    return false;
+};
+
+const visitors: Visitors = {
+    $: { scope: true },
```

Comment:
?

Reply:
This might actually not be necessary. Originally, I was going to use the `NodePath` scope mechanisms to detect whether a piece of code is within the `tmpl` or `generateMarkup` functions. However, I ended up rolling my own mechanism for that (`isWithinFn`). I'll drop this config and see if it effects the number of passing tests.

Reply:
Confirmed and config removed.